h/services/user.py

```diff
@@ -56,7 +62,55 @@ def fetch(self, userid_or_username, authority=None):
             username = parts['username']
             authority = parts['domain']
 
-        return self._cached_fetch(username, authority)
+        # The cache is keyed by (username, authority) tuples.
+        cache_key = (username, authority)
+
+        if cache_key not in self._cache:
+            self._cache[cache_key] = (self.session.query(User)
+                                      .filter_by(username=username)
+                                      .filter_by(authority=authority)
```

Comment:
The two `filter_by` queries could I think be combined into a single `filter_by(username=username, authority=authority)`. I don't know if this makes a difference to the SQL generated?

Reply:
I've just copy + pasted what we had before. They would be the same, but in the end, it doesn't really matter.

Reply:
It's actually just from a `git revert ...`